simulator/lang/lib.mjs

```diff
@@ -0,0 +1,264 @@
+import * as c from '@reach-sh/simulator-client';
+
+// nodejs compatible import
+// const c = await import('@reach-sh/simulator-client');
+
+// TODO: typescript
+
+const consensusID = -1
+const nwToken = -1
+
+Object.filter = (obj, predicate) =>
+  Object.fromEntries(Object.entries(obj).filter(predicate));
+
+
+class Scenario {
+  constructor() {
+    this.stateID = 0
+    this.participants = [];
+    this.consensus = new Consensus(new Account(-1),this);
+    this.apis = [];
+    this.views = [];
+  }
+
+  async init() {
+    // reset server
+    await c.resetServer()
+    // load the Reach program
+    const rsh = await c.load()
+    // initialize the program for the Consensus
+    await c.init()
+    const apis = await c.getAPIs()
+    const views = await c.getViews(this.stateID)
+    const l = await c.getStateLocals(this.stateID)
+
+    // setup parts
+    for (const [k,v] of Object.entries(l.l_locals)) {
+      const who = v.l_who ? v.l_who : 'Consensus'
+      const acc = new Account(v.l_acct)
+      if (who) {
+        const p = new Participant(k,acc,who,this)
+        this.participants.push(p)
+      }
+    }
+
+    // setup apis
+    for (const [k,v] of Object.entries(apis)) {
+      const who = v.a_name
+      const a = new API(k,who,this)
+      this.apis.push(a)
+    }
+
+    // setup views
+    for (const [k,v] of Object.entries(views)) {
+      const who = v.a_name
+      const vari = v.v_var
+      const tag = v.v_ty.tag
+      const contents = v.v_ty.contents
+      const v = new View(k,who,vari,tag,contents,this)
+      this.views.push(v)
+    }
+  }
+
+  async pingServer() {
+    return await c.ping();
+  }
+
+  async reset() {
+    return await c.resetServer();
+  }
+
+  async programHistory() {
+    return await c.getStates();
+  }
+
+  async getCurrentActor() {
+    const l = await c.getStateLocals(this.stateID)
+    if (l.l_curr_actor_id === consensusID) {
+      return this.consensus
+    } else {
+      return this.participants[l.l_curr_actor_id]
+    }
+  }
+
+  async newAccount() {
+    return await c.newAccount(this.stateID);
+  }
+
+  async newToken() {
+    return await c.newToken(this.stateID);
```

Comment:
You should consider have the names mimic the names in the `stdlib`, so this would be `launchToken`